src/doc/rustc/src/platform-support/x86_64-pc-cygwin.md

```diff
@@ -0,0 +1,40 @@
+# `x86_64-pc-cygwin`
+
+**Tier: 3**
+
+Windows targets supporting Cygwin and MSYS2.
+
+## Target maintainers
+
+- [Berrysoft](https://github.com/Berrysoft)
+
+## Requirements
+
+This target is cross compiled. Different `target_env` needs different linker:
+- `x86_64-pc-cygwin`: `x86_64-pc-cygwin-gcc`
+- `x86_64-pc-cygwin-msys2`: `x86_64-pc-msys-gcc`
+
+The difference between two targets are small.
+The `target_os` of these targets are `cygwin`, and they are `unix`.
+
+To gain high performance, users are recommended to use *more* native targets, e.g., `x86_64-pc-windows-*`.
+
+## Building the target
+
+For cross-compilation I recommend my fork of [llvm](https://github.com/Berrysoft/llvm-project/tree/dev/cygwin). It fixes the LLVM codegen on importing external global variables from DLLs.
```

Comment:
Are you already upstreaming these changes, and does current HEAD LLVM work or is the bug blocking?

Reply:
The current LLVM works with bug. It compiles successfully all the time, and the binary works if you write `println!("Hello world")`, but doesn't work if you link a 64-bit external variable in other dll.

I have opened a PR for upstream: https://github.com/llvm/llvm-project/pull/121439

I think it's not blocking this PR.